README.md

```diff
@@ -2,23 +2,79 @@
 
 This is a Helm plugin giving your a preview of what a `helm upgrade` would change.
 It basically generates a diff between the latest deployed version of a release
-and a `helm upgrade --debug --dry-run`
+and a `helm upgrade --debug --dry-run`. This can also be used to compare two 
+revisions/versions of your helm release.
 
 <a href="https://asciinema.org/a/105326" target="_blank"><img src="https://asciinema.org/a/105326.png" /></a>
 
 ## Usage
 
 ```
-$ helm diff [flags] RELEASE CHART
+The Helm Diff Plugin
+
+* Shows a diff explaing what a helm upgrade would change:
+    This fetches the currently deployed version of a release
+  and compares it to a local chart plus values. This can be 
+  used visualize what changes a helm upgrade will perform.
+
+* Shows a diff explaing what had changed between two revisions:
+    This fetches previously deployed versions of a release
+  and compares them. This can be used visualize what changes 
+  were made during revision change.
+
+Usage:
+  diff [command]
+
+Available Commands:
+  revision    Shows diff between revision's manifests
+  upgrade     visualize changes, that a helm upgrade could perform
```

Comment:
Restructring the command in the proposed way would be backward incompatible. I would very much would like to avoid that. Or at minimum find a clever way to alias the old CLI behaviour.
Otherwise this PR will break existing scripts and pipelines where helm is used.

I'm still liking the idea to maintain a single command `helm diff [FROM] [TO]` where `FROM` and `TO`  are more generic.

Something like this:
```
# latest revision with local chart
helm diff RELEASE local-dir/
#latest revision with latest chart from stable
helm diff RELEASE stable/postgresql
#revision 12 with latest revison
helm diff RELEASE@12
#revision 12 with revision 14
helm diff RELEASE@12 @14
```

What do you think?

Reply:
> Restructring the command in the proposed way would be backward incompatible.

It is an valid point, that we have to discuss. Those changes were proposed for the following reasons
* To have resemblance with `helm`'s usage patterns/generic usage pattern
  eg:

  ```
  # diff on upgrade
  helm upgrade [RELEASE] [CHART] [flags]
  helm diff upgrade  [RELEASE] [CHART] [flags]
  
  # diff on rollback (available as a separate pull request #24)
  helm rollback [flags] [RELEASE] [REVISION]
  helm diff rollback [flags] [RELEASE] [REVISION]
  
  # diff on revision
  helm diff revision [flags] [RELEASE] REVISION
  ```
* Features can be modular, clean and would be easy add new features. 
   For instance `rollback`

  In case these points are not appealing, we could work on it.

Reply:
hi @databus23,
any updates/suggestions on this? 

I couldn't understand the below statement
> Otherwise this PR will break existing scripts and pipelines where helm is used

could you elaborate so that it will be helpful for me to fix. 

Moreover, I couldn't find any breaking scripts and pipelines while installing the plugin. kindly correct me if am wrong.

Thanks